core/src/main/java/hudson/util/DescribableList.java

```diff
@@ -210,7 +212,12 @@ public void buildDependencyGraph(AbstractProject owner,DependencyGraph graph) {
         for (Object o : this) {
             if (o instanceof DependencyDeclarer) {
                 DependencyDeclarer dd = (DependencyDeclarer) o;
-                dd.buildDependencyGraph(owner,graph);
+                try {
+                    dd.buildDependencyGraph(owner,graph);
+                } catch (RuntimeException e) {
+                    String projectName = owner == null ? "unknown" : owner.getFullDisplayName();
+                    LOGGER.log(Level.SEVERE, "Failed to build dependency graph for " + projectName,e);
```

Comment:
For log messages, better to use `getFullName`. Or simply

    LOGGER.log(Level.SEVERE, "Failed to build dependency graph for " + owner, e);

which will contain the same information (and more) anyway.